identity/client/src/pages/groups/detail/roles/AssignRolesModal.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import { FC, useEffect, useState } from "react";
+import { Tag } from "@carbon/react";
+import { UseEntityModalCustomProps } from "src/components/modal";
+import useTranslate from "src/utility/localization";
+import { useApi, useApiCall } from "src/utility/api/hooks";
+import { searchRoles, Role } from "src/utility/api/roles";
+import { TranslatedErrorInlineNotification } from "src/components/notifications/InlineNotification";
+import styled from "styled-components";
+import DropdownSearch from "src/components/form/DropdownSearch";
+import FormModal from "src/components/modal/FormModal";
+import { assignGroupRole, Group } from "src/utility/api/groups";
+
+const SelectedRoles = styled.div`
+  margin-top: 0;
+`;
+
+const AssignRolesModal: FC<
+  UseEntityModalCustomProps<
+    { id: Group["groupKey"] },
```

Comment:
We probably need a follow up issue to replace all the groupKey / roleKey with group/role id

Reply:
I agree we need to do it, just not sure we need a new issue for it. I belive it should be a part of [the integration issue](https://github.com/camunda/camunda/issues/29720), because by the time we have the endpoints I think the key to ID refactoring will have been done on the backend (at least for those endpoints), then we just adapt the frontend to match that.